src/Symfony/Component/Security/Core/Encoder/Argon2iPasswordEncoder.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Encoder;
+
+use Symfony\Component\Security\Core\Exception\BadCredentialsException;
+
+/**
+ * ArgonPasswordEncoder uses the Argon2i hashing algorithm provided by libsodium.
+ *
+ * @author Zander Baldwin
+ */
+class Argon2iPasswordEncoder extends BasePasswordEncoder
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function encodePassword($raw, $salt)
+    {
+        if ($this->isPasswordTooLong($raw)) {
+            throw new BadCredentialsException('Invalid password.');
+        }
+
+        if (PHP_VERSION_ID >= 70200) {
```

Comment:
Agree. I'll add that in today.
I'm also trying to find a definitive answer on whether `PASSWORD_ARGON2I` will be implemented for `password_hash()` which would be a better solution than `sodium_crypto_passwd_str()`, so that feature test could change.